### DISCUSSION TITLE
Request fastapi-users from itself blocks application

### DISCUSSION THREAD

**Discussion by bert2002** [fastapi-users/fastapi-users#1121] [Q&A] [answered]
Hi,
I am trying to add a `get` router to make it easier for the user verification and within I do a http post request to the official endpoint. Somehow when I issue the request, the application stops working completely. The initial request hangs and the application does not accept any new connections. Only a restart of the application helps.

E.g.:

```
import requests
from urllib.parse import urlparse
from requests.adapters import HTTPAdapter
s = requests.Session()

@router.get("/verify/{token}")
async def verify_account(token: str):
  logger.info(f'account verification requested with token %s' % (token))

  address = os.environ.get('DOMAIN') + '/auth/verify'
  data = '{ "token": "%s" }' % (token)

  s.mount('http://', HTTPAdapter())
  r = s.post(address, data=data, headers={'Content-Type': 'application/json'})
  
  if r.status_code == 200:
    logger.info(f'account verified')
  else:
    logger.info(f'verifying account failed. status: %s' % (r.status_code))

  return {"detail": "testing"}
```
The request itself looks like this:
```
curl http://localhost:8000/auth/verify/eyJhbG....
```
It works with resources on remote endpoints, but not on the application itself. Is there any limitation known?
I am assuming something like threading and the same thread cannot process it, but not sure.

Also happy for other recommendations on how to make the verification process smoothers.

Thanks,
bert

**Comment by bert2002** [comment, marked as answer]
So the reason is that I have a `async` function and use `requests`. The solution is to either use `httpx` with an async client or dont have the function run in async.

**Comment by frankie567** [comment]
Make a HTTP request to your own application is not a good idea IMO. You already have all the logic you need you can call directly, so you can save you the roundtrip to your own server. Making the `/verify` route a GET method can be as simple as this (copy-paste from FastAPI Users code):

```py
    @router.get("/verify")
    async def verify(
        request: Request,
        token: str = Body(..., embed=True),
        user_manager: BaseUserManager[User, uuid.UUID] = Depends(get_user_manager),
    ):
        try:
            return await user_manager.verify(token, request)
        except (exceptions.InvalidVerifyToken, exceptions.UserNotExists):
            raise HTTPException(
                status_code=status.HTTP_400_BAD_REQUEST,
                detail=ErrorCode.VERIFY_USER_BAD_TOKEN,
            )
        except exceptions.UserAlreadyVerified:
            raise HTTPException(
                status_code=status.HTTP_400_BAD_REQUEST,
                detail=ErrorCode.VERIFY_USER_ALREADY_VERIFIED,
            )
```
